### ISSUE TITLE
[Enhancement]: Affichage du bulletin par période

### ISSUE THREAD

**Issue by Vexcited** [PapillonApp/Papillon#39]
### Description du problème

Voir le post dans le serveur Discord : <https://discord.com/channels/1014931881906675712/1197921133396234332>

---

On ne peut pas avoir nos bulletins depuis Papillon !

### Description de l'amélioration

Ajouter un support pour récupérer son bulletin par période dans le menu des notes (j'imagine ?)

### Contexte supplémentaire

Pawnote ne supporte pas encore cette fonctionnalité, voir la feuille de route : <https://github.com/orgs/LiterateInk/projects/1/views/1?pane=issue&itemId=55336858>

**Comment by ntillier** [comment]
Oui, je pense que c'est possible, même s'il y a deux onglets pour les anciens bulletins et ceux de cette année je crois.

Pour les anciens bulletins, Pronote fait 2 requêtes:
- Une première pour la navigation, je ne sais pas si elle est obligatoire:
```json
{
  "session": "...",
  "numeroOrdre": "...",
  "nom": "Navigation",
  "donneesSec:" {
    "_Signature_": { "onglet": 227 },
    "donnees": { "onglet": 227, "ongletPrec": 198 }
  }
}
```
et la réponse ne contient aucune données visiblement
- La seconde fait la requête pour les bulletins. Le nom est `DocumentsATelecharger`, et les données contiennent deux booléens: `avecCompetences` et `avecNotes`. Pour moi, et `avecCompetences` est `true` et `avecNotes` est `false`.
La réponse contient dans les `donnes` deux objets: `listeBulletinsBIA` et `listeCategories`.
Pour `listeCategories`, la valeur (clé: `V`) est une liste d'objets du type
```json
{
  "L": "Divers",
  "N": "0", // ou une chaîne de caractères
  "couleur": "#FFA500"
}
```
Et pour `listeBulletinsBIA`, c'est pareil, avec une liste de valeurs du type:
```json
{
  "annee": 2021,
  "ident": "une longue chaîne, je ne sais pas trop à quoi elle correspond",
  "libelle": "Bulletin de notes",
  "libelleAnnee": "2021-2022",
  "libelleClasse": "210",
  "libellePeriodeNotation": "Semestre 1",
  "nomFichier":  "PRENOM Nom-Semestre 1-210-N-6931.pdf"
}
```
En espérant que ça puisse t'aider, les noms sont assez explicites 😅

**Comment by chevillardanael** [comment]
Hello ! Des nouvelles sur la fonctionnalité ?

**Comment by ecnivtwelve** [comment]
@Vexcited

**Comment by Vexcited** [comment]
> Hello ! Des nouvelles sur la fonctionnalité ?

Hey, cette fonctionnalité n'est pas encore présente sur Pawnote malheuresement, je l'ai mise sur ma feuille de route pour la `0.21.0`

**Comment by LeGeek01** [comment]
la release 0.21.0 de Pawnote inclus cette fonctionnalité, il faut maintenant inclure ça dans papillon

**Comment by Skythrew** [comment]
Je vais essayer de m'occuper de ça si vous voulez ^^ ça m'intéresse bien :+1:
